tests/currency_network/test_gas_costs_currency_network.py

```diff
@@ -3,18 +3,56 @@
  for example because of a different solidity version.
  The tests are meant to exhibit unexpected increase in gas costs.
  They are not meant to enforce a limit.
+
+ The tests also show what gas limit could be used for certain transactions
+ These values differ from the gas costs as some gas can be reimbursed at the end of a transaction
+ e.g. for freeing storage.
+ These values are to be used in the clientlib for gas limit of transactions.
+
+ Every values are rounded up to the thousands
  """
 import pytest
 from tldeploy.core import deploy_network
 
 from ..conftest import EXTRA_DATA, EXPIRATION_TIME, get_gas_costs, report_gas_costs
 
+GAS_COST_TRANSFER_0_MEDIATOR = 61_000
+GAS_COST_TRANSFER_1_MEDIATOR = 94_000
+GAS_COST_TRANSFER_2_MEDIATORS = 128_000
+GAS_COST_TRANSFER_3_MEDIATORS = 162_000
+GAS_COST_FIRST_TL_REQUEST = 79_000
+GAS_COST_SECOND_TL_REQUEST = 45_000
+GAS_COST_FIRST_TL = 331_000
+GAS_COST_UPDATE_TL = 40_000
+GAS_COST_REDUCE_TL_LIMITS = 66_000
+GAS_COST_CLOSE_TL_NO_TRANSFER = 55_000
+GAS_COST_CLOSE_TL_TRIANGULAR_TRANSFER_2_MEDIATORS = 114_000
+GAS_COST_CLOSE_TL_TRIANGULAR_TRANSFER_4_MEDIATORS = 181_000
+GAS_COST_CANCEL_TL_UPDATE = 20_000
+
+GAS_LIMIT_TRANSFER_0_MEDIATOR = GAS_COST_TRANSFER_0_MEDIATOR
+GAS_LIMIT_TRANSFER_1_MEDIATOR = GAS_COST_TRANSFER_1_MEDIATOR
+GAS_LIMIT_TRANSFER_2_MEDIATORS = GAS_COST_TRANSFER_2_MEDIATORS
+GAS_LIMIT_TRANSFER_3_MEDIATORS = GAS_COST_TRANSFER_3_MEDIATORS
+GAS_LIMIT_FIRST_TL_REQUEST = GAS_COST_FIRST_TL_REQUEST
+GAS_LIMIT_SECOND_TL_REQUEST = GAS_COST_SECOND_TL_REQUEST
+GAS_LIMIT_FIRST_TL = 361_000
```

Comment:
Why does this value differ, nothing can be freed here, no?

Reply:
We delete the trustlineRequest

Reply:
ah okay